src/stores/video.ts

```diff
@@ -141,6 +141,71 @@ export const useVideoStore = defineStore('video', () => {
     alertStore.pushAlert(new Alert(AlertLevel.Success, `Stopped recording stream ${streamName}.`))
   }
 
+  const extractThumbnail = (): Promise<string> => {
+    return new Promise((resolve, reject) => {
+      const video = document.getElementById('mainDisplayStream') as HTMLVideoElement | null
+
+      if (!video) {
+        reject('Video element not found')
+        return
+      }
+
+      if (video.readyState >= 2) {
+        try {
+          const canvas = document.createElement('canvas')
+          canvas.width = 640
+          canvas.height = 360
+          const ctx = canvas.getContext('2d')
+          if (!ctx) {
+            throw new Error('2D context not available')
+          }
+          ctx.drawImage(video, 0, 0, canvas.width, canvas.height)
+          const thumbnailBase64 = canvas.toDataURL('image/png', 0.6)
+          resolve(thumbnailBase64)
+        } catch (error: unknown) {
+          if (error instanceof Error) {
+            reject('Failed to draw video on canvas: ' + error.message)
+          } else {
+            reject('An unexpected error occurred')
+          }
+        }
+      } else {
+        video.addEventListener(
+          'loadeddata',
+          () => {
+            try {
+              const canvas = document.createElement('canvas')
+              canvas.width = 640
+              canvas.height = 360
+              const ctx = canvas.getContext('2d')
+              if (!ctx) {
+                throw new Error('2D context not available')
+              }
+              ctx.drawImage(video, 0, 0, canvas.width, canvas.height)
+              const thumbnailBase64 = canvas.toDataURL('image/png', 0.6)
+              resolve(thumbnailBase64)
+            } catch (error: unknown) {
+              if (error instanceof Error) {
+                reject('Failed to draw video on canvas: ' + error.message)
+              } else {
+                reject('An unexpected error occurred')
+              }
+            }
+          },
+          { once: true }
+        )
+
+        video.addEventListener(
+          'error',
+          () => {
+            reject('Error when loading video data')
+          },
+          { once: true }
+        )
+      }
+    })
+  }
```

Comment:
To be honest, I'm not the biggest fan of having this thumbnail generating logic here. The last commit on the widgets, in special the `VideoPlayer`, were more in the direction of reducing the logic in them (making them "dumber") than the opposite.

This also binds an important logic of the recording pipeline to a widget, which could or not be instantiated.

Have you tried the idea of generating the thumbnail based on the first chunk, as previously discussed? Or maybe we can add to the document a video element that is not displayed, used just for this logic?

Edit: from what I'm seeing, this would not work with multiple video streams, right?